src/diagonal.works/b6/api/functions/math.go

```diff
@@ -7,35 +7,41 @@ import (
 	"diagonal.works/b6/api"
 )
 
-func divide(a api.Number, b api.Number, context *api.Context) (api.Number, error) {
-	switch b := b.(type) {
-	case api.IntNumber:
-		switch a := a.(type) {
-		case api.IntNumber:
+func divide(context *api.Context, a api.Number, b api.Number) (api.Number, error) {
+	if a, ok := a.(api.IntNumber); ok {
+		if b, ok := b.(api.IntNumber); ok {
 			return api.IntNumber(int(a) / int(b)), nil
-		case api.FloatNumber:
-			return api.FloatNumber(float64(a) / float64(b)), nil
-		}
-	case api.FloatNumber:
-		switch a := a.(type) {
-		case api.IntNumber:
-			return api.FloatNumber(float64(a) / float64(b)), nil
-		case api.FloatNumber:
-			return api.FloatNumber(float64(a) / float64(b)), nil
 		}
+		return api.FloatNumber(float64(a) / float64(b.(api.FloatNumber))), nil
+	}
+	if b, ok := b.(api.IntNumber); ok {
```

Comment:
float number ?

Reply:
if i m right and i caught this i should get a peer bonus or a beer at some point

Reply:
ah nvm i see it, u re converting on b...no beer for me